src/dao/dependency/npm.js

```diff
@@ -0,0 +1,80 @@
+const BaseDao = require('~/src/dao').BaseDao
+const NpmDependencies = require('~/src/models/dependencies/Npm')
+const cleanModel = require('windbreaker-service-util/models/util/cleanModel')
+
+let dao
+
+/**
+ * Helper function for generating query strings
+ *
+ * @param { Object } knex - the knex instance
+ * @param { String } key - the key to look for
+ * @param { Field } field - the jsonb field to search
+ *
+ * NOTE: field needs to be "manually" injected into the template field because
+ * knex will wrap the value in quotes if handled injected normally (thus causing
+ * the query to be malformed
+ */
+function _jsonKeyExistsIn (knex, key, field) {
+  return knex.raw(`${field}::jsonb ->> ? is not null`, [ key ])
+}
+
+function _formatDependencies (data) {
+  return {
+    dependencies: JSON.stringify(data.dependencies || {}),
```

Comment:
Should we just store a null (not sure what the right term is in postgres land) if these are empty? We are going to have a lot of fields that simply store `{}` and never a actually store anything else.

Reply:
Storing `null` or `undefined` is fine. Will change.